### DISCUSSION TITLE
Passing CNI_ARGS through Podman restful APIs.

### DISCUSSION THREAD

**Discussion by yihuaf** [containers/podman#14651] [Q&A]
Hi guys, I am trying to figure out how to use a custom CNI with the podman restful APIs and passing in CNI_ARGS at the same time. If using podman directly, I would run

```
CNI_ARGS=KEY=VALUE podman run --net=<my cni plugin> <image>
```

Is there a way to achieve the same effect with the restful api?

**Comment by Luap99** [comment]
Do you need to set different args for each container?
If not, it should work if you just set the env for the podman service.

**Reply by yihuaf** [reply]
Yes I would love to have different args for each container. Is there a way to do it? It doesn't have to be the CNI_ARGS. I just need a way to pass a few key value pairs down to CNI if possible.

**Reply by Luap99** [reply]
This is possible yet but sounds like a good RFE.
Are you using the libpod api endpoints or the docker compat ones?

**Reply by yihuaf** [reply]
I am looking at the libpod api. Is there a way to do it currently in the API?
